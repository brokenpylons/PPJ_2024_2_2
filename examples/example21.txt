var n = 10;
var factorial = 1;

for (var i = 1, factorial - 1) {
     factorial = factorial * i;
}

print(factorial);
