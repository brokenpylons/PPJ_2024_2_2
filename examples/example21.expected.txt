define("var") variable("n") assign("=") real("10") term(";") define("var") variable("factorial") assign("=") real("1") term(";") for("for") lparen("(") define("var") variable("i") assign("=") real("1") to(",") variable("factorial") minus("-") real("1") rparen(")") begin("{") variable("factorial") assign("=") variable("factorial") times("*") variable("i") term(";") end("}") print("print") lparen("(") variable("factorial") rparen(")") term(";") 
